pilot/pkg/model/validation.go

```diff
@@ -246,6 +250,27 @@ func (l Labels) Validate() error {
 	return errs
 }
 
+// validateTagKey checks that a string is valid as a Kubernetes label name.
+func validateTagKey(k string) error {
```

Comment:
Would it be possible for us to use k8s.io/apimachinery/pkg/util/validation.IsQualifiedName() instead of this function?  We already vendor in that package.